specification/azurelargeinstance/AzureLargeInstance.Management/models.tsp

```diff
@@ -0,0 +1,437 @@
+  @doc("Specifies the size of an empty data disk in gigabytes.")
+  diskSizeGB?: int32;
+
+  @doc("""
+Specifies the logical unit number of the data disk. This value is used to
+identify data disks within the VM and therefore must be unique for each data
+disk attached to a VM.
+""")
+  @visibility("read")
+  lun?: int32;
+}
+
+@doc("Specifies the operating system settings for the Azure Large Instance.")
+model OsProfile {
+  @doc("Specifies the host OS name of the Azure Large Instance.")
+  computerName?: string;
+
+  @doc("This property allows you to specify the type of the OS.")
+  @visibility("read")
+  osType?: string;
+
+  @doc("Specifies version of operating system.")
+  @visibility("read")
+  version?: string;
+
+  @doc("Specifies the SSH public key used to access the operating system.")
+  sshPublicKey?: string;
+}
+
+@doc("Specifies the network settings for the Azure Large Instance disks.")
+model NetworkProfile {
+  @doc("Specifies the network interfaces for the Azure Large Instance.")
+  networkInterfaces?: IpAddress[];
+
+  @doc("Specifies the circuit id for connecting to express route.")
+  @visibility("read")
+  circuitId?: string;
+}
+
+@doc("Specifies the IP address of the network interface.")
+model IpAddress {
+  @doc("Specifies the IP address of the network interface.")
+  @projectedName("csharp", "ipAddressValue")
+  ipAddress?: string;
+}
+
+@doc(""" 
+The active state empowers the server with the ability to forcefully terminate
+and halt any existing processes that may be running on the server
+""")
+model ForceState {
+  @doc("Whether to force restart by shutting all processes.")
+  @projectedName("csharp", "forceStateValue")
+  forceState?: AzureLargeInstanceForcePowerState;
+}
+
+@doc("Tags field of the AzureLargeInstance instance.")
+model Tags {
+  @projectedName("csharp", "tagsValue")
+  @doc("Tags field of the AzureLargeInstance instance.")
+  tags?: Record<string>;
+}
+
+#suppress "@azure-tools/typespec-azure-resource-manager/arm-resource-provisioning-state" "legacy API"
+@doc("Describes the properties of an AzureLargeStorageInstance.")
+model AzureLargeStorageInstanceProperties {
+  @doc("Specifies the AzureLargeStorageInstance unique ID.")
+  azureLargeStorageInstanceUniqueIdentifier?: string;
+
+  @doc("Specifies the storage properties for the AzureLargeStorage instance.")
+  storageProperties?: StorageProperties;
+}
+
+@doc("described the storage properties of the azure large storage instance")
+model StorageProperties {
+  @doc("State of provisioning of the AzureLargeStorageInstance")
+  provisioningState?: ProvisioningState;
+
+  @doc("the offering type for which the resource is getting provisioned")
+  offeringType?: string;
+
+  @doc("the storage protocol for which the resource is getting provisioned")
+  storageType?: string;
+
+  @doc("the kind of storage instance")
+  generation?: string;
+
+  @doc("the hardware type of the storage instance")
+  hardwareType?: string;
```

Comment:
Shouldn't this be of `AzureLargeInstanceHardwareTypeNamesEnum` instead of `string`?